src/darwin/Framework/CHIP/CHIPDevice.h

```diff
@@ -49,6 +72,91 @@ NS_ASSUME_NONNULL_BEGIN
                 maxInterval:(uint16_t)maxInterval
               reportHandler:(void (^)(NSArray * _Nullable value, NSError * _Nullable error))reportHandler
     subscriptionEstablished:(nullable void (^)(void))subscriptionEstablishedHandler;
+
+/**
+ * Read attribute in a designated attribute path
+ *
+ * @param completion  response handler will receive either value or error. value will be an NSArray object with NSDictionary
+ * elements. Each NSDictionary will have "endpointId", "clusterId", "attributeId", "status" and "data" keys. "endpointId",
```

Comment:
Dictionary can add new keys in the future without breaking compatibility and that applies to data marshaling between iOS and primary resident as well. That said, I agree it is a good idea to replace attribute/command path elements with path type objects because they are unlikely to change in the future.